### PR TITLE
Update upload scripts to have 1-curation-file assets and naming

### LINKED ISSUE #22: Folder upload to codeocean data asset.
Issue: 
Since curation is organized by sessions and recording, we may end up with multiple curation files in each curation folder. (see [here](https://github.com/AllenNeuralDynamics/ephys-sorting-manual-curation/tree/main/behavior_717121_2024-06-15_10-00-58/experiment1_Record%20Node%20104%23Neuropix-PXI-100.ProbeA_recording1/kilosort2_5)) They can be from different users curating the same kilosort result or even from different results of kilosort pipeline (e.g. version with and without bad channel removal). 

And with current data asset upload rule, each curation asset will have all versions of curation.json (from different users, for different sorting results), making it hard to be used by a pipeline. (e.g. https://codeocean.allenneuraldynamics.org/capsule/1846086/tree) 

@alejoe91 and I talked about different ways to solve this issue. One would be to modify the data asset content. Will it be possible to not upload the not-modified old curation files when creating a new data asset for a new/modified curation? And will it be possible to also name the curation data asset with the suffix appened to curation file? For exmaple, 'curated-opto-ZhixiaoSu' instead of 'curated'?

### PR DESCRIPTION
**Pull request by alejoe91**
Fixes #22 

@jtyoung84 @mekhlakapoor this PR refactors the upload scripts in the following way:

- The `upload_new_asset.py` generates uploads a different folder for each curation file. This is done by modifying the process_name to reflect the naming given by the user (see [here](https://github.com/AllenNeuralDynamics/ephys-sorting-manual-curation/blob/upload-script/upload_scripts/upload_utils.py#L170-L172)). 

As an example, the curation file: 

`ecephys_713591_2024-03-01_11-18-28/experiment1_Record Node 104#Neuropix-PXI-100.ProbeA_recording1/kilosort2_5/curation_KanghoonJ.json` 
will be uploaded and registered as: 
`ecephys_713591_2024-03-01_11-18-28_curated-KanghoonJ_{commit-datetime}`

- The `upload_all_assets.py` (which can be triggered manually), will upload and re-register all assets (in case we decide this is a better way to go)

Also update `aind-data-schema` to the latest version

Let me know what you guys think!

### CODE REVIEW

File: upload_scripts/upload_utils.py [hunk 1]
```diff
@@ -21,12 +20,50 @@
...
+from aind_data_schema_models.organizations import Organization
+from aind_data_schema_models.platforms import Platform
+from aind_data_schema_models.pid_names import PIDName
 from botocore.exceptions import ClientError
 
+_INVESTIGATORS_GH_TO_NAME_MAP = {
+    "alakunina": "Anna Alakunina",
+    "ZhixiaoSu": "Sue Su",
+    "KanghoonJ": "Kanghoon Jung",
+    "alejoe91": "Alessio Buccino",
```
Comment (jtyoung84) [inline upload_scripts/upload_utils.py]: We should host this in AWS Parameter Store or AWS Secrets Manager

Comment (jtyoung84) [reply upload_scripts/upload_utils.py]: Or an env variable

**Review by jtyoung84 — CHANGES_REQUESTED** [review]
- If the upload all assets is expected to run once, then I think it will be safer to just ask an admin to run the script locally rather than adding a new github action.
- Instead of hardcoding the _INVESTIGATORS_GH_TO_NAME_MAP, you can assume that is's stored as an Env Var string and parse it like `_INVESTIGATORS_TO_NAME_MAP=json.loads(os.genv("INVESTIGATORS_GH_TO_NAME_MAP"))`

File: upload_scripts/upload_utils.py [hunk 2]
```diff
@@ -21,12 +20,50 @@
...
+}
```
Comment (alejoe91) [inline upload_scripts/upload_utils.py]:
```suggestion
_INVESTIGATORS_GH_TO_NAME_MAP = json.loads(os.getenv(INVESTIGATORS_GH_TO_NAME_MAP
 ))

```

File: upload_scripts/upload_utils.py [hunk 3]
```diff
@@ -1,4 +1,3 @@
-import argparse
 import json
 import os.path
```
Comment (alejoe91) [inline upload_scripts/upload_utils.py:3]:
```suggestion
import os
import os.path
```

File: upload_scripts/requirements.txt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-aind-data-schema==0.26.5
+aind-data-schema==0.38.5
```
Comment (jtyoung84) [inline upload_scripts/requirements.txt]: Is it possible to use 1.0.0 now?

**Review by dyf — APPROVED** [review]

File: .github/workflows/upload_new_asset.yml [hunk 5]
```diff
@@ -19,12 +19,12 @@ jobs:
           fetch-depth: 2
```
Comment (jtyoung84) [inline .github/workflows/upload_new_asset.yml:21]: In the secrets section above, you'll need to add the env var like:

```
INVESTIGATORS_GH_TO_NAME_MAP: ${{ secrets.INVESTIGATORS_GH_TO_NAME_MAP }}
```

Comment (alejoe91) [reply .github/workflows/upload_new_asset.yml:21]: done in last commit

File: upload_scripts/requirements.txt (hunk 4 above)
Comment (alejoe91) [reply upload_scripts/requirements.txt]: done in last commit

**Comment by alejoe91** [comment]
@jtyoung84 I made the suggested changes!

**Review by jtyoung84 — APPROVED** [review]

**Pull request merged**